src/decode.rs

```diff
@@ -9982,13 +9980,13 @@ unsafe fn decode_b(
                 .c2rust_unnamed
                 .c2rust_unnamed_0
                 .comp_type = COMP_INTER_NONE as libc::c_int as uint8_t;
-            if !seg.is_null() && (*seg).ref_0 > 0 as libc::c_int {
+            if seg.map(|seg| seg.ref_0 > 0).unwrap_or_default() {
                 b
                     .c2rust_unnamed
                     .c2rust_unnamed_0
                     .r#ref[0 as libc::c_int
-                    as usize] = ((*seg).ref_0 - 1 as libc::c_int) as int8_t;
-            } else if !seg.is_null() && ((*seg).globalmv != 0 || (*seg).skip != 0) {
+                    as usize] = seg.as_ref().unwrap().ref_0 as i8 - 1;
```

Comment:
I think this can be an `if let Some(seg) = seg && seg.ref_0 > 0` so get rid of the `seg.unwrap()`.

Reply:
Oooh good idea, done!